ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/AstSanitizer.java

```diff
@@ -0,0 +1,283 @@
+              l -> new NodeLocation(
+                  l.getLineNumber(),
+                  l.getColumnNumber() +  "INSERT INTO".length()
+              )
+          )
+      );
+      if (target.getDataSourceType() != DataSourceType.KSTREAM) {
+        throw new KsqlException(
+            "INSERT INTO can only be used to insert into a stream. "
+                + target.getName().toString(FormatOptions.noEscape()) + " is a table.");
+      }
+      return Optional.of(
+          new InsertInto(
+              node.getLocation(),
+              node.getTarget(),
+              (Query) ctx.process(node.getQuery()),
+              node.getPartitionByColumn()
+          )
+      );
+    }
+
+    @Override
+    protected Optional<AstNode> visitAllColumns(
+        final AllColumns allColumns,
+        final StatementRewriter.Context<Void> ctx
+    ) {
+      allColumns.getSource().ifPresent(
+          src -> throwOnUnknownNameOrAlias(dataSourceExtractor, src)
+      );
+      return Optional.empty();
+    }
+
+
+    @Override
+    protected Optional<AstNode> visitSingleColumn(
+        final SingleColumn singleColumn,
+        final StatementRewriter.Context<Void> ctx
+    ) {
+      if (singleColumn.getAlias().isPresent()) {
+        selectItemIndex++;
+        return Optional.empty();
+      }
+      final ColumnName alias;
+      final Expression expression = ctx.process(singleColumn.getExpression());
+      if (expression instanceof ColumnReferenceExp) {
+        final ColumnRef name = ((ColumnReferenceExp) expression).getReference();
+        if (dataSourceExtractor.isJoin()
+            && dataSourceExtractor.getCommonFieldNames().contains(name.name())) {
+          alias = name.source()
+              .map(q -> q.name() + "_" + name.name().name())
+              .map(ColumnName::of)
+              .orElseGet(name::name);
+        } else {
+          alias = name.name();
+        }
+      } else if (expression instanceof DereferenceExpression) {
+        final DereferenceExpression dereferenceExp = (DereferenceExpression) expression;
+        final String dereferenceExpressionString = dereferenceExp.toString();
+        alias = ColumnName.of(replaceDotFieldRef(
+            dereferenceExpressionString.substring(
+                dereferenceExpressionString.indexOf(KsqlConstants.DOT) + 1)));
+      } else {
+        alias = ColumnName.generatedColumnAlias(selectItemIndex);
+      }
+      selectItemIndex++;
+      return Optional.of(
+          new SingleColumn(singleColumn.getLocation(), expression, Optional.of(alias))
+      );
+    }
+
+    private static String replaceDotFieldRef(final String input) {
+      return input
+          .replace(KsqlConstants.DOT, "_")
+          .replace(KsqlConstants.STRUCT_FIELD_REF, "__");
+    }
+
+    private DataSource<?> getSource(
+        final SourceName name,
+        final Optional<NodeLocation> location
+    ) {
+      final DataSource<?> source = metaStore.getSource(name);
```

Comment:
We should break the connection between the parser and the metastore in this PR.  Let's move the checking of whether a source is a known source or not into the _execution_ path, i.e. into the engine and out of the parser.

Reply:
done